### PR TITLE
Fixed typo and grammar in README.md

### PR DESCRIPTION
**Pull request by Saurabh-Mudgal**
Summary of resolved issues/ changes:

1. 'our' typed twice under **Join us** heading in README.md --> 'our ' is no longer repeated, typo fixed
2. verb form and comma related errors under **Memory Safety** heading in README.md --> grammar fixed
3. missing article and hyphen related errors under **Generics** heading in README.md --> grammar fixed
4. punctuation and preposition related errors under **Why build Carbon?** heading in README.md --> grammar fixed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -234,8 +234,8 @@ and with a smooth evolutionary path.
 ## Memory safety
 
 Safety, and especially
-[memory safety](https://en.wikipedia.org/wiki/Memory_safety), remain key
-challenges for C++ and something a successor language needs to address. Our
+[memory safety](https://en.wikipedia.org/wiki/Memory_safety) remains a key
```
Comment (jonmeow) [inline README.md]:
```suggestion
[memory safety](https://en.wikipedia.org/wiki/Memory_safety), remains a key
```

I think the comma is still needed here.

File: README.md [hunk 2]
```diff
@@ -57,12 +57,12 @@ GitHub will autolink `img`, but won't produce a link when `href="#"`.
...
 C++ remains the dominant programming language for performance-critical software,
 with massive and growing codebases and investments. However, it is struggling to
-improve and meet developers' needs outlined above, in no small part due to
+improve and meet developers' needs, as outlined above, in no small part due to
 accumulating decades of technical debt. Incrementally improving C++ is
 [extremely difficult](/docs/project/difficulties_improving_cpp.md), both due to
 the technical debt itself and challenges with its evolution process. The best
 way to address these problems is to avoid inheriting the legacy of C or C++
-directly, and instead start with solid language foundations like a
+directly, and instead, start with solid language foundations like
```
Comment (jonmeow) [inline README.md]:
```suggestion
directly, and instead start with solid language foundations like
```

I think this is correct as-is.

**Review by jonmeow — COMMENTED** [review]
Thanks! So you know, we got a separate fix for `our GitHub forum` so you may see that vanish from the PR when you merge with trunk.

**Comment by Saurabh-Mudgal** [comment]
Thanks for the review! I have applied the suggestions.

File: README.md [hunk 3]
```diff
@@ -57,12 +57,12 @@ GitHub will autolink `img`, but won't produce a link when `href="#"`.
...
 C++ remains the dominant programming language for performance-critical software,
 with massive and growing codebases and investments. However, it is struggling to
-improve and meet developers' needs outlined above, in no small part due to
+improve and meet developers' needs, as outlined above, in no small part due to
 accumulating decades of technical debt. Incrementally improving C++ is
 [extremely difficult](/docs/project/difficulties_improving_cpp.md), both due to
 the technical debt itself and challenges with its evolution process. The best
 way to address these problems is to avoid inheriting the legacy of C or C++
-directly, and instead start with solid language foundations like a
+directly, and instead start with solid language foundations like
```
Comment (chandlerc) [inline README.md:65]: This reads a bit off to me. I feel like "modern generics system" needs the "a", although the other entries in the list don't.

So I wonder if we should keep the "a", or should adjust this to "modern generics" (dropping "system")?

I'm happy to have this be a follow-up though, so going to go ahead and approve and merge to get the other improvements.

**Review by chandlerc — APPROVED** [review]
Really nice, thanks. Approving and merging. I think there is a follow-up question on the phrasing of one sentence though below that I'd like to not lose track of.

**Pull request merged**